### DISCUSSION TITLE
Can't move a programmatically created group on first click

### DISCUSSION THREAD

**Discussion by JiahuiChen99** [fabricjs/fabric.js#8492] [Q&A] [answered]
Hi there 👋 ,

I can't move move a group when it's first created programmatically.

1. Originally, I have an object, this object has a click handler attached to it. 
2. Once the object is clicked I create a new object and grouped them together.
3. Need to click the group again to move it.

```js
const onPointSelected = (object1) => {
    // Draw a circle around the selected point
    const object2 = new fabric.Circle({
      originX: 'center',
      originY: 'center',
      left: calibrationPoint.target.left,
      top: calibrationPoint.target.top,
      radius: 6,
    })

    // Group circle
    let group= new fabric.Group([object1, object2], {...someOptions})

    console.log(canvas.current.getActiveObject()) // AO is object1
    canvas.current.add(group)
    canvas.current.setActiveObject(pointGroup)
    console.log(canvas.current.getActiveObject()) // AO is the group
  }
```

I've tried setting the group as the canvas ActiveObject but it doesn't work. 
I'd like to move and drag the group around on object1 click. To achieve the expected behaviour, I need to re-click the group.

**Comment by ShaMan123** [comment, marked as answer]
sounds like setCoords

**Reply by JiahuiChen99** [reply]
Tried to call setCoords method after the group creation, but nothing changed. I just want to move the point around.

This works:
1. Point creation
2. Move point when dragged

What I want to do, doesn't work:
1. Point creation
2. Point click, create another object and group it
4. Move point following the first click I mentioned in the step above.

Current behaviour:
1. Point creation
2. Point click, create another object and group it
3. Click again to move

**Reply by ShaMan123** [reply]
I see.
I have in mind 2 possibilities:
1. Create the point as a group with the point in it. Then when you select the point what actually happens under the hood is that the group is selected. Adding objects is trivial, the group remains selected.
2. I don't remember if I've exposed a method for this behavior in an open PR. You can mess around with canvas state by setting `canvas._target = group`. That might work

**Reply by JiahuiChen99** [reply]
It seems that the 2nd method doesn't work for my issue. Had to use the hacky way to solve it (the first one). Instead of creating a second object and grouping them togehter on selection. I create a group. 
It is solved now, but I hope it would be more intuitive to achieve that :eyes: 
Thank you for the help @ShaMan123 ! :100:
